pkg/machines/reducers.es6

```diff
@@ -226,20 +226,34 @@ function storagePools(state, action) {
         const { connectionName, pools } = action.payload;
 
         const newState = Object.assign({}, state);
-        newState[connectionName] = {};
+        if (!(connectionName in newState))
+            newState[connectionName] = {};
+        else
+            newState[connectionName] = Object.assign({}, state[connectionName]);
 
-        // Array of strings (pool names)
-        pools.forEach(poolName => {
-            newState[connectionName][poolName] = []; // will be filled by UPDATE_STORAGE_VOLUMES
-        });
+        // Delete pools from state that are not in the payload
+        for (var poolCurrent in newState[connectionName]) {
+            if (!pools.includes(poolCurrent)) {
+                delete newState[connectionName][poolCurrent];
+            }
+        }
+
+        // Add new pools to state
+        for (var i in pools) {
+            let poolName = pools[i];
+            if (!(poolName in newState[connectionName])) {
+                newState[connectionName][poolName] = [];
+            }
+        }
 
         return newState;
     }
     case UPDATE_STORAGE_VOLUMES: {
         const { connectionName, poolName, volumes } = action.payload;
-
         const newState = Object.assign({}, state);
+
         newState[connectionName] = Object.assign({}, newState[connectionName]);
+        newState[connectionName][poolName] = Object.assign({}, state[connectionName][poolName]);
         newState[connectionName][poolName] = volumes;
```

Comment:
This doesn't look intended - the value from the previous line (which this PR adds) immediately gets overwritten by this existing line.

Reply:
I want to immutably update the dictionary under poolName, though I had to copy the object first to not overwrite it. I  guess I am understanding something wrong :/